stack/apps/push7_button/push7_button.c

```diff
@@ -112,55 +120,83 @@ static void switch_state(APP_STATE_t new_state)
     DPRINT("entering a new state: %d", new_state);
     current_app_state = new_state;
     sensor_manager_set_transmit_state(new_state == OPERATIONAL_STATE || new_state == TEST_STATE);
-    sensor_manager_set_test_mode(new_state == TEST_STATE);
-    if (new_state == CONFIGURATION_STATE) {
+    if (new_state == SENSOR_CONFIGURATION_STATE) {
         sensor_manager_get_sensor_states(sensor_enabled_state_array);
     }
 }
 
-static void display_state(bool state)
-{
-    if (state)
-        led_flash_white();
-}
+static void display_state(bool state) { start_led_flash(state ? 1 : 2); }
 
 static void operational_input_event_handler(input_type_t i, bool mask) { }
 
-static void configuration_input_event_handler(input_type_t i, bool mask)
+static void sensor_configuration_input_event_handler(input_type_t i, bool mask)
 {
+    if (i == STATE_COUNTER_EVENT || i == HALL_EFFECT_EVENT)
+        return;
+
     if (current_buttons_state == NO_BUTTON_PRESSED) {
-        if(max_buttons_state == NO_BUTTON_PRESSED)
+        if (max_buttons_state == NO_BUTTON_PRESSED)
             return;
-        if (max_buttons_state != prev_max_buttons_state) {
-            DPRINT("showing the state of %d",max_buttons_state );
-            display_state(sensor_enabled_state_array[max_buttons_state]);
 
-        } else {
+        if (max_buttons_state == prev_max_buttons_state) {
```

Comment:
I don't get this max_buttons_state

Reply:
It's there to determine if multiple buttons were pressed at the same time.